### ISSUE TITLE
Allows use of the ontrack parameter type in declarative pipeline syntax

### ISSUE THREAD

**Issue by SansGuidon** [jenkinsci/ontrack-plugin#40]
We can use ontrack parameters in declarative syntax using the following code, so we retrieve dynamic parameters, but we still need to write a lot of ugly code to achieve that (example below).

```
properties([
    parameters([
        [
            $class: 'OntrackChoiceParameterDefinition', 
            description: '', 
            dsl: "ontrack.branch('APS', 'release').standardFilter(count: 5)",
            name: 'RELEASE_NUMBER', 
            valueProperty: 'name',
            sandbox: false
        ]
    ])
])
pipeline {
    agent any
   /* parameters {
        // no way at this moment to have dynamic ontrack parameters here
    }*/
    stages {
        stage('Example') {
            steps {
	         echo 'deployment of config for ' + params.RELEASE_NUMBER
            }
        }
    }
}
```

Hopefully, a solution exists, simple to implement by taking inspiration from other plugins. 
So, the aim of this issue is to improve support of declarative pipeline and allow dynamic parameters as below :

```
pipeline {
    agent any
    parameters {
        ontrackChoiceParam(name: 'RELEASE_NUMBER', description: '...', dsl: '...', valueProperty: '...')
    }
    stages {
        stage('Example') {
            steps {
	         echo 'deployment of config for ' + params.RELEASE_NUMBER
            }
        }
    }
}
``` 

How ? 

In order to have per-plugin parameters recognised by Jenkins, it's needed to define a specific Symbol in classes that extend ParameterDefinition https://github.com/jenkinsci/ontrack-plugin/search?utf8=%E2%9C%93&q=ParameterDescriptor+&type= 

Example here with the another plugin (credentials) where a Symbol annotation is defined 
https://github.com/jenkinsci/credentials-plugin/blob/master/src/main/java/com/cloudbees/plugins/credentials/CredentialsParameterDefinition.java#L115

More info about this problem and solution for credentials plugin, if it's helpful : 
https://issues.jenkins-ci.org/browse/JENKINS-44588
https://github.com/jenkinsci/credentials-plugin/pull/86/files

🚧 I'm currently working on a solution, and it seems to fill the needs perfectly  ☕️ 

Cheers
Morgan

**Comment by SansGuidon** [comment]
@dcoraboeuf you can assign it to me if you like 😉

**Comment by dcoraboeuf** [comment]
Here you go :)

**Comment by SansGuidon** [comment]
@dcoraboeuf tried to push a new branch but was not allowed, not sure if I'm allowed to contribute to this repo, can you confirm ? :)

**Comment by dcoraboeuf** [comment]
You have to fork the repository first and then to create a PR. That'd be the fastest. We can also see if you're interested in becoming a direct contributor.

**Comment by SansGuidon** [comment]
ok ;) will do

**Comment by dcoraboeuf** [comment]
Available in 2.32.3
